components/flow-orchestration-framework/org.wso2.carbon.identity.flow.execution.engine/src/main/java/org/wso2/carbon/identity/flow/execution/engine/FlowExecutionService.java

```diff
@@ -66,15 +67,19 @@ public FlowExecutionStep executeFlow(String tenantDomain, String applicationId,
                                          String flowType, Map<String, String> inputs) throws FlowEngineException {
 
         FlowExecutionStep step;
+        FlowExecutionContext context = null;
         try {
-            FlowExecutionContext context;
             if (StringUtils.isBlank(flowId)) {
                 // No flowId present hence initiate the flow.
                 context = FlowExecutionEngineUtils.initiateContext(tenantDomain, applicationId, flowType);
             } else {
                 context = FlowExecutionEngineUtils.retrieveFlowContextFromCache(flowType, flowId);
             }
-            Optional.ofNullable(inputs).ifPresent(inputs1 -> context.getUserInputData().putAll(inputs1));
+
```

Comment:
any reason for changing this

Reply:
Context should be final to be used here which cannot be done when using at catch.